source/use-cases/sensitive-data-encryption.txt

```diff
@@ -257,25 +246,25 @@ access the key for automatic encryption and decryption.
 The client requires the following configuration values to generate a new
 data encryption key:
 
-- The local master key or AWS KMS master key access settings.
+- The locally-managed master key or AWS KMS master key access settings.
 
 - A MongoDB connection string that authenticates on a running server.
 
 - The key vault namespace (database and collection).
 
 Follow the steps below to generate a single data encryption key from the
-local master key.
+locally-managed master key.
 
 .. tabs-drivers::
    :hidden: true
    
    .. tab::
       :tabid: java-sync
        
-      You can also download the `complete code example on GitHub
-      <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-csfle-data-encryption-key/DataEncryptionKeyGenerator.java>`_.
+      You can also download the `complete code example
+      <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-json-schema-helper-and-json/DataEncryptionKeyGenerator.java>`_.
 
-.. include:: /includes/steps/create-data-encryption-key.rst
+.. include:: /includes/steps/fle-create-data-encryption-key.rst
```

Comment:
A few thoughts:

- For step 2, can we point to https://mongodb.github.io/mongo-java-driver/3.11/javadoc/com/mongodb/AutoEncryptionSettings.Builder.html#kmsProviders(java.util.Map) so users have a reference for why we're doing a map of maps of maps?
- For step 3, note that the base64 encoded key ID is required for automatic field level encryption
- For step 4, we're using the built in Document to string. I think Jeff Yemin provided a better example that returns the proper JSON - can we use that instead?

Reply:
Step 2: sure, I'll add a link. Other than it being the documented interface, is there any other info that explains why it's a map of maps?

Step 3: sure; the code snippet also contains a comment and variable name that reflects that they need to use the base64 key id, but I can see how it would be helpful to mention earlier. Will update.

Step 4: I'm on the fence about this proposed addition. My understanding of the extended json section you're asking for would explain both extended json and each parameter. I'd like to separate the document from function or feature documentation and instead maintain its focus as a tutorial. I can definitely make a note and include a link to mention the ejson representation. Let me know what you think @rkumar-mongo